src/main/java/com/android/volley/toolbox/DiskBasedAsyncCache.java

```diff
@@ -102,13 +114,14 @@ public void failed(Throwable exc, Void ignore) {
         } catch (IOException e) {
             VolleyLog.e(e, "Failed to read file %s", file.getAbsolutePath());
             closeChannel(channel, "IOException");
+            remove(key, emptyCallback);
```

Comment:
Same here, I think we should wait until the callback returns before moving on

Reply:
For here, should I use a CompletableFuture to wait until it returns? My hesitation there is that we'd then have to deal with more exceptions, when we're already in the catch block.

Reply:
Actually, I can just call get's callback in the callback instead of doing anything with a future. But in this case, does it necessarily matter? Since the key will be mapped to nothing anyway?

Reply:
I see your point, but it'd feel safer if the get callback only returns once the work is fully complete, including when it returns nothing. This way, the caller knows when the async work is complete (and the callback method is called `onGetComplete`)

Reply:
yeah I agree. I ended up just caliing onGetComplete in the onComplete callback.